pkg/services/datasources/service/cache.go

```diff
@@ -108,3 +127,14 @@ func idKey(id int64) string {
 func uidKey(orgID int64, uid string) string {
 	return fmt.Sprintf("ds-orgid-uid-%d-%s", orgID, uid)
 }
+
+func (dc *CacheServiceImpl) canQuery(user *user.SignedInUser, ds *datasources.DataSource) error {
+	guardian := dc.dsGuardian.New(user.OrgID, user, *ds)
```

Comment:
does not need to be changed now but if `New` is always called with `user.OrgID` could it be simplified to just `New(user, *ds)`?

I would really simplify SignedInUser interfacing :innocent:

Reply:
Yup, for sure can. I'll leave it for another PR as this one has already grown too huge 🙈